blockchain/reactor.go

```diff
@@ -231,19 +231,22 @@ FOR_LOOP:
 					break SYNC_LOOP
 				} else {
 					bcR.pool.PopRequest()
+					// TODO: use ApplyBlock instead of Exec/Commit/SetAppHash/Save
```

Comment:
see handshake branch. it basically puts the ExecBlock, Commit, and setting of state root in one function